cognite/client/data_classes/transformations_common.py

```diff
@@ -0,0 +1,145 @@
+        return TransformationDestination(type="files")
+
+    @staticmethod
+    def labels():
+        """To be used when the transformation is meant to produce labels."""
+        return TransformationDestination(type="labels")
+
+    @staticmethod
+    def relationships():
+        """To be used when the transformation is meant to produce relationships."""
+        return TransformationDestination(type="relationships")
+
+    @staticmethod
+    def data_sets():
+        """To be used when the transformation is meant to produce data sets."""
+        return TransformationDestination(type="data_sets")
+
+    @staticmethod
+    def raw(database: str = "", table: str = ""):
+        """To be used when the transformation is meant to produce raw table rows.
+
+        Args:
+            database (str): database name of the target raw table.
+            table (str): name of the target raw table
+
+        Returns:
+            TransformationDestination pointing to the target table
+        """
+        return RawTable(type="raw", database=database, table=table)
+
+
+class RawTable(TransformationDestination):
+    def __init__(self, type: str = None, database: str = None, table: str = None):
+        super().__init__(type=type)
+        self.database = database
+        self.table = table
+
+    def __hash__(self):
+        return hash((self.type, self.database, self.table))
+
+    def __eq__(self, obj):
+        return isinstance(obj, RawTable) and hash(obj) == hash(self)
+
+
+class OidcCredentials:
+    def __init__(
+        self,
+        client_id: str = None,
+        client_secret: str = None,
+        scopes: str = None,
+        token_uri: str = None,
+        audience: str = None,
+        cdf_project_name: str = None,
+    ):
+
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.scopes = scopes
+        self.token_uri = token_uri
+        self.audience = audience
+        self.cdf_project_name = cdf_project_name
+
+    def dump(self, camel_case: bool = False) -> Dict[str, Any]:
+        """Dump the instance into a json serializable Python data type.
+
+        Args:
+            camel_case (bool): Use camelCase for attribute names. Defaults to False.
+
+        Returns:
+            Dict[str, Any]: A dictionary representation of the instance.
+        """
+        ret = {
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+            "scopes": self.scopes,
+            "token_uri": self.token_uri,
+            "audience": self.audience,
+            "cdf_project_name": self.cdf_project_name,
+        }
+        if camel_case:
+            return {utils._auxiliary.to_camel_case(key): value for key, value in ret.items()}
+        return ret
+
+
+class TransformationBlockedInfo:
+    def __init__(self, reason: str = None, created_time: Optional[int] = None, time: Optional[int] = None):
```

Comment:
excluding `time`  here gives a decoding error but we also don't want to support it as it is a deprecated field. Could we workaround it?

Reply:
pop it from the dict here? https://github.com/cognitedata/cognite-sdk-python/pull/868/files#diff-4f82056777939cc794f3fd64dcbedce16cbcdd19ae67a7a4d1377957f5b524a8R123